rerun_cpp/src/rerun/c/sdk_info.h

```diff
@@ -0,0 +1,14 @@
+/// Returns the version of the Rerun C SDK.
+///
+/// This should match the string returned by `rr_version_string` (C) or `rerun::version_string` (C++).
+/// If not, the SDK's binary and the C header are out of sync.
+#define RERUN_SDK_HEADER_VERSION "0.18.0-alpha.1"
+
+/// Major version of the Rerun C SDK.
+#define RERUN_SDK_HEADER_VERSION_MAJOR 0
+
+/// Minor version of the Rerun C SDK.
+#define RERUN_SDK_HEADER_VERSION_MINOR 18
+
+/// Patch version of the Rerun C SDK.
+#define RERUN_SDK_HEADER_VERSION_PATCH 0
```

Comment:
We should also have a macro for easily comparing to these. Something like


```C
/// Is the Rerun library version greater or equal to this?
///
/// Example usage:
/// ```
/// #if RERUN_VERSION_GE(0, 18, 0)
///    // Use features from Rerun 0.18
/// #endif
/// ```
#define RERUN_VERSION_GE(major, minor, patch)            \
    ((major) == RERUN_SDK_HEADER_VERSION_MAJOR ?         \
        ((minor) == RERUN_SDK_HEADER_VERSION_MINOR ?     \
            (patch) <= RERUN_SDK_HEADER_VERSION_PATCH :  \
            (minor) <= RERUN_SDK_HEADER_VERSION_MINOR) : \
        (major) <= RERUN_SDK_HEADER_VERSION_MAJOR)
```

And then add a quick test of it under `rerun_cpp/tests`